File: frontend/console/src/api/action.js
import Axios from "@/http";

export function createAction(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/action', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveAction(actionId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/action/${ actionId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateAction(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/action', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteAction(actionId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/action/${ actionId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteActionWithFlag(actionId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/action/${ actionId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllActions() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/action/all').then(res => {
            resolve(res)
        })
    })
}

export function queryAction (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/action/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

File: little-bee/controller/router/action.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_ACTION = "/config/action"
const URL_ROUTER_ACTION_BY_ID = "/config/action/:action-id"
const URL_ROUTER_ACTION_QUERY = "/config/action/query"
const URL_ROUTER_ACTION_ALL = "/config/action/all"

func ActionRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_ACTION, config.CreateAction)
	r.GET(URL_ROUTER_ACTION_BY_ID, config.RetrieveAction)
	r.PUT(URL_ROUTER_ACTION, config.UpdateAction)
	r.DELETE(URL_ROUTER_ACTION_BY_ID, config.DeleteAction)
	r.PUT(URL_ROUTER_ACTION_BY_ID, config.DeleteActionWithFlag)
	r.POST(URL_ROUTER_ACTION_QUERY, config.QueryAction)
	r.GET(URL_ROUTER_ACTION_ALL, config.GetAllActions)
	return r
}